### PR TITLE
HomeWork2

### PR DESCRIPTION
**Pull request by Rchum0**

### CODE REVIEW

File: src/main/java/Problem3/Book.java [hunk 1]
```diff
@@ -16,10 +16,24 @@ public Book(String title, String author) {
     // copy constructor
     public Book(Book anotherBook) {
         // homework
+        this.title = anotherBook.title;
```
Comment (jmkelly6) [inline src/main/java/Problem3/Book.java:19]: Only this.id = anotherbook.id is needed.

File: src/main/java/Problem3/BookFiction.java [hunk 2]
```diff
@@ -25,5 +30,11 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0) {
+            int lateFee = numOfDaysPastDue * lateFeePerDayInDollar;
+            return lateFee;
+        }else {
```
Comment (jmkelly6) [inline src/main/java/Problem3/BookFiction.java:36]: Suggestion for the future: you don't need to add the else{} branch; instead you can just return 0 outside of the if-statement since the function will always return 0 if it does not pass the "if" conditions.

File: src/main/java/Problem3/BookRomance.java [hunk 3]
```diff
@@ -22,5 +24,11 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0) {
+            int lateFee = numOfDaysPastDue * lateFeePerDayInDollar;
+            return lateFee;
+        }else {
```
Comment (jmkelly6) [inline src/main/java/Problem3/BookRomance.java:30]: Same thing here, see comment in BookFiction.java

File: src/main/java/Problem3/MovieAction.java [hunk 4]
```diff
@@ -21,6 +24,15 @@ public int getLateFeeInDollar() {
...
     public int calcLateFees(int numOfDaysPastDue) {
-        // homework
+        //
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
+            int lateFee = numOfDaysPastDue * lateFeePerDayInDollar;
+            return lateFee;
+        } else if (numOfDaysPastDue >= 5) {
+            int lateFee = (2 * (numOfDaysPastDue * lateFeePerDayInDollar));
+            return lateFee;
+        } else {
```
Comment (jmkelly6) [inline src/main/java/Problem3/MovieAction.java:34]: See comments on else{} branch in BookFiction.java

File: src/main/java/Problem3/MovieComedy.java [hunk 5]
```diff
@@ -21,5 +24,11 @@ public int getLateFeeInDollar() {
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
         // homework
+        if (numOfDaysPastDue > 0) {
+            int lateFee = numOfDaysPastDue * lateFeePerDayInDollar;
+            return lateFee;
+        }else {
```
Comment (jmkelly6) [inline src/main/java/Problem3/MovieComedy.java:30]: See else{} recommendations

File: src/test/java/Problem2Test.java [hunk 6]
```diff
@@ -9,11 +9,15 @@ public void testRemoveDuplicates() {
                 // DO NOT REMOVE OR CHANGE THE GIVEN TEST CASES!
                 {0, 0, 1, 1, 1, 2, 2, 4, 4, 6}
                 // add more test cases
+                , {10, 10, 20, 20, 20, 30, 30, 40, 60}
+                , {1, 1, 1, 1, 2, 2, 3, 4}
```
Comment (jmkelly6) [inline src/test/java/Problem2Test.java:13]: Recommendation for future assignments: add edge cases (ex. when an object is NULL or empty) to verify your functions work for those too.

File: src/main/java/Problem3/MovieAction.java [hunk 7]
```diff
@@ -21,6 +24,15 @@ public int getLateFeeInDollar() {
 
     @Override
     public int calcLateFees(int numOfDaysPastDue) {
-        // homework
+        //
+        if (numOfDaysPastDue > 0 && numOfDaysPastDue < 5) {
```
Comment (jmkelly6) [inline src/main/java/Problem3/MovieAction.java:28]:
```suggestion
        int fee = lateFeePerDayInDollar * numOfDaysPastDue;
        return (numOfDaysPastDue >= 5) ? 2 * fee : fee;
```

**Review by pdgetrf — APPROVED** [review]
LGTM
